smithy-build/src/main/java/software/amazon/smithy/build/plugins/RemoveSharedShapesPlugin.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.build.plugins;
+
+import software.amazon.smithy.build.PluginContext;
+import software.amazon.smithy.build.SmithyBuildPlugin;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.node.Node;
+import software.amazon.smithy.model.shapes.ModelSerializer;
+import software.amazon.smithy.model.transform.ModelTransformer;
+
+public final class RemoveSharedShapesPlugin implements SmithyBuildPlugin {
```

Comment:
Why not call this RemoveTraitShapes / "remove-trait-shapes"?

Reply:
Fixed.  This evolved a bit and I didn't update the naming.

Reply:
This should probably be a `ProjectionTransformer` and not a `SmithyBuildPlugin` since it needs to modify a model and not necessarily create its own artifacts.

Reply:
This has been reworked and is now a transformer.